User search field

ActiveChooN opened the pull request:
<!---
Copyright (C) 2020 Intel Corporation

SPDX-License-Identifier: MIT
-->

<!-- Raised an issue to propose your change (https://github.com/opencv/cvat/issues).
It will help avoiding duplication of efforts from multiple independent contributors.
Discuss your ideas with maintainers to be sure that changes will be approved and merged.
Read the [CONTRIBUTION](https://github.com/opencv/cvat/blob/develop/CONTRIBUTING.md)
guide. -->

<!-- Provide a general summary of your changes in the Title above -->

### Motivation and context
<!-- Why is this change required? What problem does it solve? If it fixes an open
issue, please link to the issue here. Describe your changes in detail, add
screenshots. -->

![изображение](https://user-images.githubusercontent.com/8168727/97449777-158d0a00-1943-11eb-9268-00ed148c8553.png)

This PR provide user search field for assignee field for the task. It should increase UX with user search by username and increase loading speed for cvat.org.

### How has this been tested?
<!-- Please describe in detail how you tested your changes.
Include details of your testing environment, and the tests you ran to
see how your change affects other areas of the code, etc. -->

Manually

### Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply.
If an item isn't applicable by a reason then ~~explicitly strikethrough~~ the whole
line. If you don't do that github will show incorrect process for the pull request.
If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
~~- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly~~
~~- [ ] I have added tests to cover my changes~~
~~- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

bsekachev commented:
Could you please merge develop to CI passes?

bsekachev commented:
It looks like some unit tests of cvat-core & cypress tests are broken

ActiveChooN commented:
@bsekachev, all tests was fixed

bsekachev commented:
Now each new character creates new HTTP request. How about debouncing a user input? 0.5-1 seconds I suppose would be appropriate.
![Screenshot from 2020-11-10 12-44-33](https://user-images.githubusercontent.com/40690378/98657070-84f6f680-2352-11eb-8ea7-0ad6b2348b0f.png)

bsekachev commented:
It is not very obvious how to remove assignee with current UI element. To do that I need to remove the text and click outside the input field (enter doesn't work for example). I suppose it might be more user-friendly. 
One more approach which probably better I see is adding X icon at the right of input for the purpose.

bsekachev commented:
@ActiveChooN 
On other points the patch is nice and it significantly accelerates the user interface. Let's fix left comments, resolve conflicts and merge it

bsekachev commented:
@dvkruchinin Hi, on CI we get ``Failed to connect to Chrome, retrying in 1 second (attempt 23/62)``. As far as I remember you've fixed something like that. Doesn't the fix work? Maybe try to find another ways or workarounds? It is not a rare issue

ActiveChooN commented on cvat-core/.eslintrc.js:
```diff
@@ -4,6 +4,7 @@
 
 module.exports = {
     env: {
+        amd: true,
```
Eslint warn about require usage without that flag.

bsekachev replied:
Just set ``node`` to true

dvkruchinin commented:
> @dvkruchinin Hi, on CI we get `Failed to connect to Chrome, retrying in 1 second (attempt 23/62)`. As far as I remember you've fixed something like that. Doesn't the fix work? Maybe try to find another ways or workarounds? It is not a rare issue

Hi,
It was more of a test. Since in the discussions, cypress users wrote that someone this fix helped, someone did not. Yes, I'll keep looking. And I want to believe that this error has become less common.

ActiveChooN commented:
@bsekachev 

> 
> 
> It is not very obvious how to remove assignee with current UI element. To do that I need to remove the text and click outside the input field (enter doesn't work for example). I suppose it might be more user-friendly.
> One more approach which probably better I see is adding X icon at the right of input for the purpose.

Added `Enter` button for empty search request

> I suppose the styles also applicable for the new element

Strange margins here, removed styles
